source/parser/pattern-matching.civet

```diff
@@ -438,14 +439,25 @@ function processDeclarationCondition(condition, rootCondition, parent): void
   return unless condition.type is "DeclarationCondition"
   ref := makeRef()
 
-  { decl, bindings } := condition.declaration
+  { decl, bindings } := condition.declaration as DeclarationStatement
   // TODO: Add support for `let` and `const` declarations with multiple bindings in conditions
   binding := bindings[0]
   { pattern, suffix, initializer, splices, thisAssignments } := binding
 
+  grandparent := condition.parent?.parent
+  children :=
+    // Check that the declaration is a plain assignment (no pattern-matching) and the immediate grandchild of an `if` or `while`
+    // More complex conditions (triggered by pattern matching or `until`/`unless`) don't need double parens
+    // @ts-ignore Just because binding.pattern might not have a type at runtime doesn't mean it's unsafe
+    if binding.pattern.type is "Identifier" and (grandparent?.type is "IfStatement" or grandparent?.type is "WhileStatement")
+      // steal the parens from the parent, forcing them to double up
+      [condition.parent.children.0, ref, initializer, condition.parent.children.-1]
+    else
+      [ref, initializer]
+
   Object.assign condition,
     type: "AssignmentExpression"
-    children: [ref, initializer]
+    children: children
```

Comment:
Maybe worth wrapping the object in braces to cleanup `children`, `pattern`, and `ref`.